src/query/expression/src/block.rs

```diff
@@ -92,24 +92,16 @@ impl<T: BlockMetaInfo> BlockMetaInfoDowncast for T {
         if boxed.as_any().is::<T>() {
             unsafe {
                 // SAFETY: `is` ensures this type cast is correct
-                let raw_ptr = Box::into_raw(boxed) as *const dyn BlockMetaInfo;
-                return Some(std::ptr::read(raw_ptr as *const Self));
+                let raw_ptr = Box::into_raw(boxed) as *mut dyn BlockMetaInfo;
+                let typed_ptr = raw_ptr as *mut Self;
+                return Some(*Box::from_raw(typed_ptr));
```

Comment:
it'll be more consice if we write:
```
let downcast_box = std::mem::transmute::<Box<_>, Box< Self>>(boxed);
*downcast_box
```

Reply:
Good idea! BTW, I want to discuss, what about removing unsafe code by writing somthing like `boxed.into_any().downcast()`

Reply:
it may be not able to take ownership from Box<Any>, but we can clone it anyway.